src/core/ColorPicker/ColorSwatch.tsx

```diff
@@ -30,14 +30,28 @@ export const ColorSwatch = React.forwardRef<HTMLDivElement, ColorSwatchProps>(
     const { color, style, onClick, isActive, className, ...rest } = props;
     useTheme();
 
-    const c =
+    const [colorString, setColorString] = React.useState(() =>
       typeof color === 'string'
         ? color
-        : getColorValue(color).toHslString(true);
-    const _style =
-      color && getWindow()?.CSS?.supports?.(`--swatch-color: ${c}`)
-        ? { '--swatch-color': c, ...style }
-        : { backgroundColor: c, ...style };
+        : getColorValue(color).toHslString(true),
+    );
+
+    React.useEffect(() => {
+      const newString =
+        typeof color === 'string'
+          ? color
+          : getColorValue(color).toHslString(true);
+
+      setColorString(newString);
+    }, [color]);
```

Comment:
This is new. Was the style not updating without this useEffect?

Reply:
The problem was that getColorValue(color).toHslString(true) way too many times of every render for every swatch.  Now it only gets called if the color value changes.

Reply:
why not use `useMemo`?

Reply:
good point ... I was too fast to reach for useEffect... I will update.

Reply:
changed